js/hview.js

```diff
@@ -406,39 +409,41 @@ $(function () {
                     if (valueF == valueI) {
                         const valueU = valueI >>> 0;
                         let valueHex = "";
+                        let onChangeCallback = () => { };
+
                         if (p.name.search(/color$/i) >= 0) {
                             valueHex = toHex(valueU, 8);
                             const colorHexOption = optionProtoType.cloneNode()
                             colorHexOption.value = 'color-hex'
                             colorHexOption.innerHTML = "#" + valueHex
                             selectTag.appendChild(colorHexOption)
-                        } else {
-                            valueHex = toHex(valueU);
-                            const falgsHexOption = optionProtoType.cloneNode()
-                            falgsHexOption.value = 'falgs-hex'
-                            falgsHexOption.innerHTML = "0x" + valueHex
-                            selectTag.appendChild(falgsHexOption)
-                        }
-                        if (valueHex) {
+
                             colorWellDiv = divProtoType.cloneNode()
                             colorWellDiv.classList.add(CLS_COLORWELL)
-
-                            selectTag.onchange = () => {
-                                const myVal = "" + selectTag.val();
-                                if (myVal.startsWith("#")) {
+                            onChangeCallback = () => {
+                                if (selectTag.value == 'color-hex') {
                                     const webColor = '#' + toHex(argb2rgba(valueU), 8);
                                     colorWellDiv.style.display = 'inline-block'
                                     colorWellDiv.style.backgroundColor = webColor
                                 } else {
                                     colorWellDiv.style.display = "none"
                                 }
-                            }
+                            };
+                            selectTag.addEventListener("change", onChangeCallback);
+
+                        } else {
+                            valueHex = toHex(valueU);
+                            const falgsHexOption = optionProtoType.cloneNode()
+                            falgsHexOption.value = 'falgs-hex'
+                            falgsHexOption.innerHTML = "0x" + valueHex
+                            selectTag.appendChild(falgsHexOption)
```

Comment:
Is this section fixing a bug, or is it simply refactoring but maintaining current behavior?

Reply:
there is a bug. The color box does not get displayed when selecting the color in values dropdown